code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -46,6 +46,11 @@
 					overlays += "unlocked"
 			else
 				overlays += "off"
+		if(istype(src, /obj/structure/closet/body_bag)) // GUESS WHAT BODY BAGS ARE CLOSETS

```

Comment:
This should be handled in an update_icon() proc of /obj/structure/closet/body_bag that calls the parent with ..()